rest/route/middleware.go

```diff
@@ -402,20 +402,42 @@ func (m *CommitQueueItemOwnerMiddleware) ServeHTTP(rw http.ResponseWriter, r *ht
 
 	// The owner of the patch can also pass
 	vars := gimlet.GetVars(r)
-	item, ok := vars["item"]
+	itemId, ok := vars["item"]
 	if !ok {
-		item, ok = vars["patch_id"]
+		itemId, ok = vars["patch_id"]
 	}
-	if !ok || item == "" {
+	if !ok || itemId == "" {
 		gimlet.WriteResponse(rw, gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
 			StatusCode: http.StatusBadRequest,
 			Message:    "No item provided",
 		}))
 		return
 	}
 
-	if projRef.CommitQueue.PatchType == commitqueue.SourceCommandLine {
-		patch, err := m.sc.FindPatchById(item)
+	cq, err := commitqueue.FindOneId(projRef.Id)
+	if err != nil || cq == nil {
+		gimlet.WriteResponse(rw, gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    "No commit queue for project found",
+		}))
+		return
+	}
+	var entry *commitqueue.CommitQueueItem
+	for _, item := range cq.Queue {
+		if item.Issue == itemId {
```

Comment:
Can you use https://github.com/evergreen-ci/evergreen/blob/9fea335a3da54159f950942e4b778838bd64b644/model/commitqueue/commit_queue.go#L139 for these kinds of loops?

Reply:
done